resources/test/tests/functional/api-tests-1.html

```diff
@@ -173,6 +173,19 @@ <h1>Sample HTML5 API Tests</h1>
         assert_throws("TEST_ERR", function() {throw e});
     }, "Test assert_throws with non-DOM-exception expected to Fail");
 
+    function FooError() {
+      return this;
+    }
+    test(function()
+    {
+        assert_throws(new FooError(), function() {throw new FooError()});
```

Comment:
Can we allow `assert_throws(FooError, ...);`? Functions also have `name` property so I think it's doable without too much changes.